### DISCUSSION TITLE
Is there a way to disassemble or assemble executable created by Pyinstaller?

### DISCUSSION THREAD

**Discussion by someziggyman** [pyinstaller/pyinstaller#6557] [Help]
Hey guys, 
Tried searching for the answer, but I'm not sure there's an exact term that matches my question.. 
Moreover, just started to familiarise with Pyinstaller.

If I know for sure that executable was created with Pyinstaller, is there a way to "disassemble" it (to a state as if --onefile was not applied)? 
For example, to be able to codesign everything needed and put it all back together to "--onefile"? 
Running MacOS, if it matters.

Really appreciate you help with it.

**Comment by rokm** [comment]
Sure, the PyInstaller packing/unpacking code is not exactly a secret, so with enough skill and motivation, you can create your own (re-)packer.

You can use the [`pyi-archive_viewer`](https://github.com/pyinstaller/pyinstaller/blob/develop/archive_viewer.py) utility to list/extract contents of embedded archive(s), or write your own unpacking tool that uses [PyInstaller's archive classes](https://github.com/pyinstaller/pyinstaller/tree/develop/PyInstaller/archive).

Creating the executable is a bit more complicated, especially on macOS, so your best bet is probably to recreate/reuse the [PyInstaller's executable assembly pipeline](https://github.com/pyinstaller/pyinstaller/blob/d0d8baec5f8fdb9f92a82dfb299031a5d159b37c/PyInstaller/building/api.py#L592-L790).

That said, if your main concern is codesigning, you might be better off using [--codesign-identity](https://pyinstaller.readthedocs.io/en/stable/usage.html#cmdoption-codesign-identity) (see [docs](https://pyinstaller.readthedocs.io/en/stable/feature-notes.html#macos-binary-code-signing)). Or using `onedir` builds with manual post-hoc signing.

**Reply by someziggyman** [reply]
Thanks a lot for your hints! Skill is a bit of a problem on this end))) Ordinary user slightly upgraded to amateur bash scripting.

My goal here is to build stand alone portable ytdlp https://github.com/yt-dlp/yt-dlp from source.. OR from their precompiled release.

To simplify this task, I thought I could codesign executables they provide here and from this stage assemble it into one executable.
Using their precompiled executable created with --onefile and codesigning ,it works locally but not on my other "clean" machine... 
<img width="492" alt="Screenshot 2022-01-31 at 19 50 40" src="https://user-images.githubusercontent.com/17865486/151846366-85fee981-5e1e-4b7e-b32a-2feca8e52c83.png">

Is there a command to assemble these files from the screenshot to --onefile?

Files on the screenshot have been created with pyinstaller, I assume without --onefile param

**Reply by bwoodsend** [reply]
There isn't. And since the source code is available I 100% recommend going down the route of just building from source. You should be able to inject the `--codesign-identity` argument [into their slightly OTT build script](https://github.com/yt-dlp/yt-dlp/blob/1c6f480160199d88b8e3d278befe6eec099b619d/pyinst.py#L44) and run it.
